File: all-submissions/0965-unique-email-addresses/solution.java
class Solution {
    public int numUniqueEmails(String[] emails) {
        HashSet<String> uniquemail = new HashSet<String>();
        for(String email : emails)
        {
            String[] mail = email.split("@");
            String localname = mail[0].replace(".","");
            int index = localname.indexOf("+");
            localname = localname.substring(0,index);
            //int index1 = mail[1].indexOf(".com");
            String domainname = mail[1];
            // domainname = domainname.replace(".","");
            // domainname = domainname.replace("+","");
            String umail = localname+domainname;
            uniquemail.add(umail);
            }
        return uniquemail.size();
        }
        
    }

File: all-submissions/0965-unique-email-addresses/solution.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        dict_email = defaultdict(set)

        def do_format(email_id: str) -> str:
            email_id = email_id.partition('+')[0].replace('.','')
            return email_id

        for email in emails:
            email_id , domain_name = email.split('@')
            actual_id = do_format(email_id)
            dict_email[domain_name].add(actual_id)
        ans = 0
        for email_ids in dict_email.values():
            ans += len(email_ids)
        return ans
